var/spack/repos/builtin/packages/lemon/package.py

```diff
@@ -16,3 +16,49 @@ class Lemon(CMakePackage):
     url = "https://lemon.cs.elte.hu/pub/sources/lemon-1.3.1.tar.gz"
 
     version("1.3.1", sha256="71b7c725f4c0b4a8ccb92eb87b208701586cf7a96156ebd821ca3ed855bad3c8")
+
+    variant("coin", default=False, description="Enable Coin solver backend")
+    variant("ilog", default=False, description="Enable ILOG (CPLEX) solver backend")
+    variant("glpk", default=True, description="Enable GLPK solver backend")
+    # soplex not mentioned in docs but shown in cmakecache
+    variant("soplex", default=False, description="Enable SOPLEX solver backend")
+
+    variant(
+        "lp_backend",
+        default="glpk",
+        description="Set default LP solver backend",
+        values=("glpk", "cplex", "clp", "soplex"),
+        multi=False,
+    )
+    variant(
+        "mip_backend",
+        default="glpk",
+        description="Set default MIP solver backend",
+        values=("glpk", "cplex", "cbc"),
+        multi=False,
+    )
+
+    depends_on("glpk", when="+glpk")
+    # LP solver
+    depends_on("glpk", when="lp_backend=glpk")
+    depends_on("cplex", when="lp_backend=cplex")
+    depends_on("clp", when="lp_backend=clp")
+    # MIP solver
+    depends_on("glpk", when="mip_backend=glpk")
+    depends_on("cplex", when="mip_backend=cplex")
+    depends_on("cbc", when="mip_backend=cbc")
+
+    def cmake_args(self):
+        spec = self.spec
+        args = []
+        args.extend(
+            [
+                "-DLEMON_ENABLE_COIN={0}".format("YES" if "+coin" in spec else "NO"),
+                "-DLEMON_ENABLE_ILOG={0}".format("YES" if "+ilog" in spec else "NO"),
+                "-DLEMON_ENABLE_GLPK={0}".format("YES" if "+glpk" in spec else "NO"),
+                "-DLEMON_ENABLE_SOPLEX={0}".format("YES" if "+soplex" in spec else "NO"),
+                "-DLEMON_DEFAULT_LP={0}".format(spec.variants["lp_backend"].value.upper()),
+                "-DLEMON_DEFAULT_MIP={0}".format(spec.variants["mip_backend"].value.upper()),
```

Comment:
There are helper methods for `CMakePackage` (see https://spack.readthedocs.io/en/latest/build_systems/cmakepackage.html#adding-flags-to-cmake) though apparently not providing the expected values.

We are moving to increased use of f-strings (https://docs.python.org/3/tutorial/inputoutput.html#fancier-output-formatting) over `format()`.

Finally, we are also moving toward use of `spec.satisfies()` for checking boolean variant values (see https://spack.readthedocs.io/en/latest/packaging_guide.html#testing-spec-constraints, especially the note in the last paragraph where use of `in spec` will include variant values of dependencies.

Reply:
Thanks for the suggestions. So, should I do the following instead:
```python
    def cmake_args(self):
        spec = self.spec
        args = []
        args.extend(
            [
                f"-DLEMON_ENABLE_COIN={spec.variants['coin'].value}",
                f"-DLEMON_ENABLE_ILOG={spec.variants['ilog'].value}",
                f"-DLEMON_ENABLE_GLPK={spec.variants['glpk'].value}",
                f"-DLEMON_ENABLE_SOPLEX={spec.variants['soplex'].value}",
            ]
        )
        return args
```

Reply:
i checked that the package builds fine. It seems cmake doesn't care if it's a yes/no or true/false.